vsintegration/src/FSharp.Editor/LanguageService/LanguageService.fs

```diff
@@ -598,10 +584,16 @@ and
         base.SetupNewTextView(textView)
 
         let textViewAdapter = package.ComponentModel.GetService<IVsEditorAdaptersFactoryService>()
-               
+
         match textView.GetBuffer() with
         | (VSConstants.S_OK, textLines) ->
             let filename = VsTextLines.GetFilename textLines
+
+
+            // CPS projects don't implement IProvideProjectSite and IVSProjectHierarchy
+            // Simple explanation:
+            //    Legacy projects have IVSHierarchy and IPRojectSite
+            //    CPS Projects and loose script files don't
```

Comment:
But could you also document why we need to implement a VS interface at all here?  It looks really odd in the code to be doing that